### PR TITLE
Add expires_on support in Az cli cred flow

### PR DESCRIPTION
**Pull request by g2vinay**
Adds `expires_on` field support in Az cli credential.

Closes https://github.com/Azure/azure-sdk-for-java/issues/37504

### CODE REVIEW

**Comment by g2vinay** [comment]
pushing unit test next.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java [hunk 1]
```diff
@@ -655,6 +655,22 @@ AccessToken getTokenFromAzureCLIAuthentication(StringBuilder azCommand) {
         return token;
     }
 
+    static OffsetDateTime getTokenExpiryOffsetDateTime(Map<String, String> objectMap) {
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java:658]: nit: maybe this is two helper methods: one to parse `expires_on`, one to parse `expiresOn`. That way you can test and detect that we are properly detecting when `expires_on` isn't present. Right now there's no way to assert which path through this method we took.

**Review by billwert — APPROVED** [review]
one small suggestion, but looks good.

File: sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java [hunk 2]
```diff
@@ -89,8 +91,22 @@ public void testValidSecret() {
...
 
+        //Epoch equivalent of "2023-10-31 21:59:10.000000" is 1698814750;
+        tokenDetails.put("expiresOn", "2023-10-31 21:59:10.000000");
+        OffsetDateTime offsetDateTime = IdentityClientBase.getTokenExpiryOffsetDateTime(tokenDetails);
+        Assertions.assertEquals(offsetDateTime.toEpochSecond(), 1698814750);
 
+        // Test the scenario with expires_on present, it should be given priority.
+        tokenDetails.put("expires_on", "1572371520");
+        offsetDateTime = IdentityClientBase.getTokenExpiryOffsetDateTime(tokenDetails);
+        Assertions.assertEquals(offsetDateTime.toEpochSecond(), 1572371520);
```
Comment (billwert) [inline sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java:109]: nit: reversed actual and expected.

File: sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java [hunk 3]
```diff
@@ -89,8 +91,22 @@ public void testValidSecret() {
...
+
+    @Test
+    public void testExpiresOnParsingAzureCli() {
+        // setup
+        Map<String, String> tokenDetails = new HashMap<>();
 
+        //Epoch equivalent of "2023-10-31 21:59:10.000000" is 1698814750;
+        tokenDetails.put("expiresOn", "2023-10-31 21:59:10.000000");
+        OffsetDateTime offsetDateTime = IdentityClientBase.getTokenExpiryOffsetDateTime(tokenDetails);
+        Assertions.assertEquals(offsetDateTime.toEpochSecond(), 1698814750);
```
Comment (billwert) [inline sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java:104]: nit: reversed actual and expected.

**Review by billwert — APPROVED** [review]
this is still marked draft but I think it looks good at this point.

**Pull request merged**